packages/flutter_tools/lib/src/flx.dart

```diff
@@ -235,7 +230,41 @@ Future<int> build(
       return result;
     }
 
-    archive.addFile(_createSnapshotFile(snapshotPath));
+    snapshotFile = _createSnapshotFile(snapshotPath);
+  }
+
+  return assemble(
+      manifestDescriptor: manifestDescriptor,
+      snapshotFile: snapshotFile,
+      assetBasePath: assetBasePath,
+      materialAssetBasePath: materialAssetBasePath,
+      outputPath: outputPath,
+      privateKeyPath: privateKeyPath
+  );
+}
+
+/// Assembles a Flutter .flx file from a pre-existing manifest descriptor
+/// and a pre-compiled snapshot.
+///
+/// This may be called by external build toolchains, so practice caution
+/// when changing this method signature (alert flutter-dev).
+Future<int> assemble({
```

Comment:
We may want to promote this file to the `lib/` dir (instead of being in `lib/src/`). `lib/` is API public, and `src/` means internal impl. It'll make it easier for us to know which APIs to be careful about.

If we do that, we should demote some things which are currently library public in this file to private.